packages/mdl-ripple/index.js

```diff
@@ -39,67 +21,123 @@ export default class MDLRipple {
     background.classList.add(Class.BACKGROUND);
     root.appendChild(background);
 
-    this.elements_ = {
-      [Identifier.ROOT]: root,
-      [Identifier.BACKGROUND]: background,
-      [Identifier.SURFACE]: surface,
-    };
-
     for (let i = 0; i < MAX_RIPPLES; i++) {
       let foreground = document.createElement('div');
       foreground.classList.add(Class.FOREGROUND);
       let foregroundCircle = document.createElement('div');
       foregroundCircle.classList.add(Class.FOREGROUND_CIRCLE);
       foreground.appendChild(foregroundCircle);
       root.appendChild(foreground);
+    }
+    return root;
+  }
 
-      this.elements_[this.get(Identifier.FOREGROUND, i)] = foreground;
-      this.elements_[this.get(Identifier.FOREGROUND_CIRCLE, i)] = foregroundCircle;
+
+  /**
+   * @param {!Element} surface
+   * @param {!Element} root
+   * @return {!MDLRipple}
+   */
+  static attachTo(surface, root) {
+    let elements = {
+      [Identifier.SURFACE]: surface,
+      [Identifier.ROOT]: root,
+      [Identifier.BACKGROUND]: root.getElementsByClassName(Class.BACKGROUND)[0]
```

Comment:
I don't have much of a problem with this personally. If needs change later it isn't like this API specifically is relied upon. While the `querySelector` method is cleaner this still works just fine and is easily understood.

As far as project maintenance though, there is the question of consistency. I think using `querySelector`/`All` APIs throughout would just keep peoples mental state vs having to interpret the API in use itself.

I'm marginally for converting to `querySelector` for that reason, but don't really care on a personal level.